### PR TITLE
Merge absolute imports into feature/abi

### PR DESCRIPTION
**Pull request by michaeldiamant**
Merges https://github.com/algorand/pyteal/pull/283 into `feature/abi` _without_ updating imports for `pyteal.ast.abi` files.

Details:
* The PR temporarily ignores I252 in `pyteal.ast.abi` to keep the diff smaller.   A subsequent PR will remove the temporary ignore and update imports.
* After applying the temporary I252 ignore, several legitimate lint errors remaining.  I fixed them by-hand.
```
flake8 docs examples pyteal scripts tests *.py
pyteal/ir/tealop.py:84:13: I252 Relative imports are banned.
pyteal/ast/__init__.py:141:1: I252 Relative imports are banned.
make: *** [flake8] Error 1
```

### CODE REVIEW

**Comment by michaeldiamant** [comment]
PR is marked as _draft_ because it _includes_ #287.  I'll mark ready for review after merging #287.  The diff will shrink.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,33 +2,79 @@
...
 
-
-
-# Filing Issues
+## Filing Issues
 
 Did you discover a bug? Do you have a feature request? Filing issues is an easy way anyone can contribute and helps us improve PyTeal. We use GitHub Issues to track all known bugs and feature requests.
 
-Before logging an issue be sure to check current issues, check the [Developer Frequently Asked Questions](https://developer.algorand.org/docs/developer-faq) and [GitHub issues][issues_url] to see if your issue is described there.
+Before logging an issue be sure to check current issues, check the [open GitHub issues][issues_url] to see if your issue is described there.
```
Comment (tzaffi) [inline CONTRIBUTING.md:10]: 👍 for removing dead link

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -2,33 +2,79 @@
...
+
+Modules may be created inside of other modules, in which case the deeper module is called a submodule or child module, and the module that contains it is called the parent module. For example, `pyteal` is the parent module to `pyteal.ast`.
+
+A sibling module is defined as a different child module of the parent module. For example, `pyteal.ast` and `pyteal.ir` are sibling modules.
+
+### Import Guidelines
+
+#### In Runtime Code
+
+When a runtime file in this codebase needs to import another module/file in this codebase, you should import the absolute path of the module/file, not the relative one, using the `from X import Y` method.
```
Comment (tzaffi) [inline CONTRIBUTING.md:46]:
```suggestion
A file in this codebase that requires another module/file in this codebase, should import the absolute path of the module/file, not the relative one, using the `from X import Y` method.
```

**Review by tzaffi — APPROVED** [review]
I'm conditionally approving with one suggestion for a non-essential wording improvement.

**Pull request merged**